src/main/java/teammates/logic/core/CoursesLogic.java

```diff
@@ -716,4 +716,9 @@ public boolean hasIndicatedSections(String courseId) throws EntityDoesNotExistEx
         return archivedCourseIds;
     }
 
+    @Deprecated
+    public List<CourseAttributes> getAllCourses() {
+        return coursesDb.getAllCourses();
+    }
```

Comment:
IMO such methods should not be added to production code for the sole purpose of supporting client scripts. See https://github.com/TEAMMATES/teammates/pull/7671#discussion_r125308911

Reply:
I'm of the same opinion, but judging from the number of methods I deleted here and the previous PR #8369 you'd know how much that is not followed.
This one I can easily work around because it's a call to an existing DB method without any modification.